src/tools/compiletest/src/runtest.rs

```diff
@@ -2898,8 +2901,14 @@ impl<'test> TestCx<'test> {
             &self.config.compare_mode,
             kind,
         );
-        if !path.exists() && self.config.compare_mode.is_some() {
-            // fallback!
+
+        if !path.exists() {
```

Comment:
Hmm, this is a bit more hard-coded than I expected, but I imagine it does the right thing.

Reply:
Maybe it's not worth making a super general mechanism here, given that we'll hopefully be removing polonius and nll "soon".

Reply:
What I expected was some kind of table where, for each compare-mode, we specify a "parent" mode (which may be `None` for nll) — and we iterate through modes, walking to their parents (and ultimately to a `None` mode) — looking for a path that exists.